host/libhackrf/src/hackrf.h

```diff
@@ -165,274 +385,816 @@ typedef struct hackrf_device hackrf_device;
+ * @return @ref HACKRF_SUCCESS on success or @ref HACKRF_ERROR_LIBUSB
+ */
 extern ADDAPI int ADDCALL hackrf_init();
 
+/**
+ * Exit libhackrf
+ * 
+ * Should be called before exit. No other libhackrf functions should be called after it. Can be safely called multiple times.
+ * @return @ref HACKRF_SUCCESS on success or @ref HACKRF_ERROR_NOT_LAST_DEVICE if not all devices were closed properly.
+ */
 extern ADDAPI int ADDCALL hackrf_exit();
 
+/**
+ * Get library version.
+ * 
+ * Can be called before @ref hackrf_init
+ * @return library version as a human-readable string
+ */
 extern ADDAPI const char* ADDCALL hackrf_library_version();
 
+
+/**
+ * Get library release.
+ * 
+ * Can be called before @ref hackrf_init
+ * @return library version as a human-readable string. 
+ */
 extern ADDAPI const char* ADDCALL hackrf_library_release();
 
+/**
+ * List connected HackRF devices
+ * 
+ * @return list of connected devices. The list shuld be deleted with @ref hackrf_device_list_free
+ */
 extern ADDAPI hackrf_device_list_t* ADDCALL hackrf_device_list();
 
+/**
+ * Open a @ref hackrf_device from a device list
+ * 
+ * @param[in] list device list to open device from
+ * @param[in] idx index of the device to open
+ * @param[out] device device handle to open
+ * @return @ref HACKRF_SUCCESS on success, @ref HACKRF_ERROR_INVALID_PARAM on invalid parameters or other @ref hackrf_error variant
+ */
 extern ADDAPI int ADDCALL hackrf_device_list_open(
 	hackrf_device_list_t* list,
 	int idx,
 	hackrf_device** device);
 
+/**
+ * Delete a previously allocated @ref hackrf_device_list list.
+ * 
+ * @param[in] list list to delete
+ */
 extern ADDAPI void ADDCALL hackrf_device_list_free(hackrf_device_list_t* list);
 
+/**
+ * Open first alaviable HackRF device
+ * 
+ * @param[out] device device handle
+ * @return @ref HACKRF_SUCCESS on success, @ref HACKRF_ERROR_INVALID_PARAM if @p device is NULL, @ref HACKRF_ERROR_NOT_FOUND if no HackRF devices are found or other @ref hackrf_error variant
+ */
 extern ADDAPI int ADDCALL hackrf_open(hackrf_device** device);
 
+/**
+ * Open HackRF device by serial number
+ * 
+ * @param[in] desired_serial_number serial number of device to open. If NULL then default to first device found.
+ * @param[out] device device handle
+ * @return @ref HACKRF_SUCCESS on success, @ref HACKRF_ERROR_INVALID_PARAM if @p device is NULL, @ref HACKRF_ERROR_NOT_FOUND if no HackRF devices are found or other @ref hackrf_error variant
+ */
 extern ADDAPI int ADDCALL hackrf_open_by_serial(
 	const char* const desired_serial_number,
 	hackrf_device** device);
 
+/**
+ * Close a previously opened device
+ * 
+ * @param[in] device device to close
+ * @return @ref HACKRF_SUCCESS on success or variant of @ref hackrf_error
+ */
 extern ADDAPI int ADDCALL hackrf_close(hackrf_device* device);
 
+/**
+ * Start receiving
+ * 
+ * Shuld be called after setting gains
```

Comment:
In this case, I think the default values of gains should be documented (possibly at the gain setting functions, and at the corresponding overview section)

Reply:
There aren't any default gains in the libhackrf API, though. The device retains the last gains that were set.

There are defaults in the firmware, but libhackrf could be talking to any version of the firmware, including any number of forked versions, so I don't think the libhackrf API documentation should be making any guarantees about what the firmware defaults are.

I think it's fine to point out that the caller _may_ want to set the gains before calling `hackrf_start_rx`, I just think the language needs to be quite careful and specific, because "should" can be taken to mean that it's required to do it before every call.